script_parameters.yml

```diff
@@ -0,0 +1,252 @@
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        docker_command: |
+          # Parameter values are added to your container as environment variables.
+          # See the script params reference for a full list.
+          echo "my_boolean = $MY_BOOLEAN"
+          echo "my_integer = $MY_INTEGER"
+          echo "my_float = $MY_FLOAT"
+          # To prevent command injection attacks, you can get shell-escaped versions of string parameters.
+          echo "my_string = $MY_STRING_SHELL_ESCAPED"
+          echo "my_file is at $MY_FILE_URL"
+          # It is generally not advisable to log your credentials.
+          # my_custom_credential generates $MY_CUSTOM_CREDENTIAL_USERNAME, $MY_CUSTOM_CREDENTIAL_PASSWORD
+          # my_aws_credential generates $MY_AWS_CREDENTIAL_ACCESS_KEY_ID, $MY_AWS_CREDENTIAL_SECRET_ACCESS_KEY
+          # my_db_credential generates $MY_DB_CREDENTIAL_USERNAME, $MY_DB_CREDENTIAL_PASSWORD
+        # You can add this if you want to prevent the created script from
+        # appearing on the scripts index page, to reduce clutter.
+        hidden: true
+      on-success:
+        - python_script
+
+    python_script:
+      action: civis.scripts.python3
+      input:
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          # Optionally, you can specify the parameter values in the `params` list
+          # instead of in the `arguments` hash, by using the `default` key.  But
+          # this does not work for credential parameters.
+          - name: my_boolean
+            type: bool
+            default: true
+          - name: my_integer
+            type: integer
+            default: 1234
+          - name: my_float
+            type: float
+            default: 0.1234
+          - name: my_string
+            type: string
+            default: this is a string
+          - name: my_file
+            type: file
+            default: 0  # ID of the file
+          - name: my_custom_credential
+            type: credential_custom
+          - name: my_aws_credential
+            type: credential_aws
+          - name: my_db_credential
+            type: credential_redshift
+        arguments:
+          my_custom_credential: 0  # ID of the credential
+          my_aws_credential: 0
+          my_db_credential: 0
+        source: |
+          # Parameter values are added to your container as environment variables.
+          # See the script params reference for a full list.
+
+          import os
+
+          # Type conversions are necessary.  Unfortunately, `bool()` does not work here.
+          my_boolean = os.environ['MY_BOOLEAN'] == 'true'
```

Comment:
Yeah it'd be worth a more explicit callout/explanation about this boolean/string thing, I've seen it confuse people over and over again